frontend/app/src/composables/blockchain/balances/index.ts

```diff
@@ -49,16 +51,29 @@ export const useBlockchainBalances = () => {
         true
       );
       const balances = BlockchainBalances.parse(result);
-      const ethBalances = balances.perAccount[Blockchain.ETH];
-
-      if (ethBalances) {
-        const addresses = [...Object.keys(ethBalances)];
-        await fetchEnsNames(addresses, ignoreCache);
-      }
       updateEth(blockchain, balances);
       updateBtc(blockchain, balances);
       updateChains(blockchain, balances);
       setStatus(Status.LOADED);
+
+      get(evmChains).forEach(chain => {
+        if (!isBlockchain(chain)) {
+          return;
+        }
+
+        const perChainBalances = balances.perAccount[chain];
+        if (!perChainBalances) {
+          return;
+        }
+
+        const addresses = [...Object.keys(perChainBalances)];
+        startPromise(
+          fetchEnsNames(
+            addresses.map(address => ({ address, blockchain: chain })),
+            ignoreCache
+          )
+        );
+      });
```

Comment:
Doesn't this do a separate request for each chain? is there any reason why we cannot do everything?